src/ExerciseCleaner.php

```diff
@@ -196,9 +202,13 @@ public function cleanFiles(array $pathList, float $targetStep = 1, bool $solutio
                     // Avoid double slashes in grep result
                     $path = substr($path, 0, -1);
                 }
-                $cmd = "grep '{$this->startTagConstant}' -Rl $path";
-                if ($suffix) {
-                    $cmd .= " | grep -v '$suffix$'";
+                if ($inputExtension) {
+                    $cmd = "find $path -name *$inputExtension";
```

Comment:
`--input-ext training` will treat _example.php.training_ but not _example.training.php_.

From an IDE point of view:
- _example.training.php_ will be considered by default as PHP;
- _example.php.training_ won't be by default and `*.php.training` must be configured as being PHP.

From an application point of view:
- _example.training.php_ could be included in the application, considered as bad PHP, duplicate PHP class or whatever;
- _example.php.training_ probably means nothing for the application and is ignored.

I prefer to have to configure the IDE rather than break the application. So, the usage of a dedicated extension seems to have sens only when this extension is truly at the end.